Source/LightsOutSln/LightsOut/Patches/ModCompatibility/Biotech/PatchWasteAtomiser.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using LightsOut.Common;
+using RimWorld;
+using Verse;
+
+namespace LightsOut.Patches.ModCompatibility.Biotech
+{
+    public class PatchWasteAtomiser : ICompatibilityPatchComponent<PatchWasteAtomiser>
+    {
+        public override string ComponentName => "Patch waste atomiser";
+        public override IEnumerable<PatchInfo> GetPatches(Type type)
+        {
+            var patches = new List<PatchInfo>
+            {
+                new PatchInfo
+                {
+                    method = GetMethod<CompAtomizer>(nameof(CompAtomizer.CompTick)),
```

Comment:
Avoid patching tickwise methods at all cost, they are more expensive than they seem.

You are likely better off patching `CompAtomizer`'s `Notify_MaterialsAdded`, `EjectContents`, and `DoAtomize` methods to cover your functionality.